core/src/main/java/org/apache/druid/java/util/http/client/pool/ResourcePool.java

```diff
@@ -177,7 +183,8 @@ private ImmediateCreationResourceHolder(
       this.unusedResourceTimeoutMillis = unusedResourceTimeoutMillis;
       this.resourceHolderList = new ArrayDeque<>();
 
-      for (int i = 0; i < maxSize; ++i) {
+      final int initializationSize = eagerInitialization ? maxSize : 1;
+      for (int i = 0; i < initializationSize; ++i) {
```

Comment:
If `eager` is false, it would create just one connection on start-up.
Would it ever create more connections or will it always remain at 1?

Reply:
It would create more connections on the go